### PR TITLE
Add test harness

### LINKED ISSUE #2: Test code examples and links
See #1 and https://github.com/rust-lang-nursery/mdBook/issues/706#issuecomment-432204598 regarding code.

See https://github.com/Michael-F-Bryan/mdbook-linkcheck regarding links.

### LINKED ISSUE #46: Broken links due to rand_distr split
## Overview

**Severity: Low**

I do not have the time and am not familiar with this codebase, but there are several broken links in the ["guide-dist" page](https://github.com/rust-random/book/blob/master/src/guide-dist.md). This is due to the fact that the distributions were split out from this crate at some point in the past into the rand_dist crate.

## Verification of Issue

Go to the documentation generated from this page and find this text:
> The Normal distribution (also known as Gaussian) simulates sampling from the Normal distribution ("Bell curve") with the given mean and standard deviation.

The word "Normal" is pointing to the link

https://rust-random.github.com/rust-random/rand/rand/distributions/struct.Normal.html

but it should point to

https://rust-random.github.io/rand/rand_distr/struct.Normal.html

## Explanation

This is because it is generated from the text
```markdown
The [`Normal`] distribution (also known as Gaussian) simulates sampling from
the Normal distribution ("Bell curve") with the given mean and standard
deviation.
```
which uses a WIkilink style link.

## Proposed Mitigation

This would need to be updated to use a full Markdown link (in the style `(text)[URI]`) to be fixed.

I have checked, and there are more instances of this on the page, probably many, so I don't want to try to fix only 1 or 2. If anyone more familiar with the project sees this, they might have a better idea which links need fixing.

### PR DESCRIPTION
**Pull request by dhardy**
Uses `doc_comment` to add a test harness for the code samples within documentation. This allows us to specify dependency versions ([not possible through mdbook](https://github.com/rust-lang/mdBook/issues/706)).

Note that not all code samples actually *run* since this uses the playground, which [currently uses an old version of `rand_pcg`](https://github.com/integer32llc/rust-playground/issues/780).

As for testing links... in theory `mdbook-linkcheck` does this, but using it on this repo generates a lot of errors like this:
```
error: Linking outside of the "root" directory is forbidden
    ┌─ update-0.8.md:190:1
    │
190 │ [`IteratorRandom::choose_stable`] was added to provide an alternative that
    │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Linking outside of the "root" directory is forbidden
```
These links (e.g. `../rand/rand/seq/trait.IteratorRandom.html#method.choose_stable`) work where deployed (and locally with a symlink to the built docs). We could replace with absolute URLs, but I suspect `linkcheck` would ignore them in that case.

... thus, I just laboriously clicked on every link to test.

Some documentation got updated here: the Sequences page, part of the Distributions page. @vks would be nice if you do a quick review.

Closes #46. Closes #2 (well, link checks are omitted — without a plan for handling cross-crate links I don't see much use for it).

### CODE REVIEW

File: src/contributing.md [hunk 1]
```diff
@@ -47,5 +47,5 @@ thus we may take considerable time to get back to you.
     
     Our works are attributed to "The Rand Project Developers". This is not a
     formal entity but merely the collection of all contributors to this project.
-    For more, see the [COPYRIGHT](COPYRIGHT) file.
+    For more, see the `COPYRIGHT` file.
```
Comment (vks) [inline src/contributing.md:50]: Why did you remove the link?

File: src/guide-dist.md [hunk 2]
```diff
@@ -111,29 +116,15 @@ Lets go over the distributions by type:
...
-probability `p` of *success* (`true`).
+-   The [`Bernoulli`] distribution simply generates a boolean where the
+    probability of sampling `true` is some constant (`Bernoulli::new(0.5)`) or
+    ratio (`Bernoulli::from_ratio(1, 6)`).
+-   The [`WeightedIndex`] distribution may be used to sample from a sequence of
+    weighted values. See the [Sequences] section.
 
-The methods [`Rng::gen_bool`] and [`Rng::gen_ratio`] are short-cuts to this
-distribution.
+Many more distributions are provided by the [`rand_distr`] crate.
```
Comment (vks) [inline src/guide-dist.md:127]:
```suggestion
Many more non-uniform distributions are provided by the [`rand_distr`] crate.
```

File: src/guide-err.md [hunk 3]
```diff
@@ -19,24 +19,15 @@ reduce to calls to [`RngCore`]'s "infallible" methods. Since most RNGs cannot
...
-    appear to be capable of sufficient precision or is too predictable, this
-    will fail.
--   [`EntropyRng`] is an abstraction over the above, falling back to the next
-    option when the first fails but ultimately failing if all sources fail
--   [`thread_rng`] seeds itself via [`EntropyRng`], thus can potentially fail
-    on its first use on each thread (though it never fails after the first use)
--   [`ReadRng`] tries to read data from its source but fails when the stream
-    ends or errors (though it retries on interrupt).
+-   [`OsRng`] is a wrapper over [`getrandom`]. "In general, on supported
+    platforms, failure is highly unlikely, though not impossible."
```
Comment (vks) [inline src/guide-err.md:23]: Why the quotes?

File: src/guide-seq.md [hunk 4]
```diff
@@ -1,16 +1,92 @@
...
+-   [`WeightedIndex`] has fast setup and `O(log N)` sampling
+-   [`WeightedAliasIndex`] has slow setup and `O(1)` sampling, thus *may* be
+    faster with a large number of samples
+
+For convenience, you may use:
+
+-   [`SliceRandom::choose_weighted`]
+-   [`SliceRandom::choose_weighted_mut`]
+
+#### Without replacement
```
Comment (vks) [inline src/guide-seq.md:59]:
```suggestion
### Without replacement
```

**Review by vks — APPROVED** [review]
Looks great! I only have some minor questions and suggestions.

File: src/contributing.md (hunk 1 above)
Comment (dhardy) [reply src/contributing.md:50]: Because it was broken. You can try it in the current version of the book. We don't publish this file, so we could only link back to the repo.

File: src/guide-err.md (hunk 3 above)
Comment (dhardy) [reply src/guide-err.md:23]: Because it's a quote from the getrandom documentation. Maybe this is too brief.
